src/dsl/dsl.jl

```diff
@@ -71,21 +72,77 @@ function address_from_expression(lhs)
 end
 
 function desugar_tildes(expr)
+    trace_ref = GlobalRef(@__MODULE__, Symbol("@trace"))
+    line_num = LineNumberNode(1, :none)
```

Comment:
@ztangent Any idea if this changes anything re: error messages? i.e., do crashes still report the right line number for `@trace` calls?

Reply:
Pretty sure the previous thing that was being done (where we were constructing `@trace(addr)` expressions as a quoted expression) also wasn't including line numbers!

Reply:
Yup, confirmed:
```julia> dump(:(@trace(call, addr)))
Expr
  head: Symbol macrocall
  args: Array{Any}((4,))
    1: Symbol @trace
    2: LineNumberNode
      line: Int64 1
      file: Symbol none
    3: Symbol call
    4: Symbol addr
```

So at the very least it's not a regression.